### ISSUE TITLE
2406 skill id invalid

### ISSUE THREAD

**Issue by mifrith** [tuya/tuya-home-assistant#613]
Hello!
Following the instructions, I created a new account and a new project. When I try to log in, I get an error: Invalid authentication.
In logs: 2021-10-21 10:33:31 ERROR (MainThread) [custom_components.tuya_v2.config_flow] Login failed: {'code': 2406, 'msg': 'skill id invalid', 'success': False, 't': 1634801611400}.
The SmartLife account is linked, the devices on the Tuya website are visible.
"Central Europe Data Center" is selected in the project settings, country code for authorization is 7.

**Comment by Actpohomoc** [comment]
Same to me. Login error (2406): skill id invalid.
Recreate project two times. 
Never find "Smart Home Devices Management" API service.
I had project before 22-09-2020. Subscribe to everything that can find related.

I think this is that moment when we have to move to ESPHome!

**Comment by mazzy89** [comment]
@zlinoliver then please mark this integration as broken because it does not even support the basic functionalities and make the repo achieved. there is no sense to leave it around because it is broken if it can't even support proper authentication and configuration.

**Comment by SebCar62** [comment]
bonjour j'ai réussi a contournez l'erreur .j'ai mis comme pays Spain a la place de switzerland. et cela a fonctionnez

**Comment by tekelectron** [comment]
This has to be the worse integration I have ever seen, it should be pulled instantly.

**Comment by cnschulz** [comment]
Yeah, this is super weird.  Although my (original) app was set to mainland china (years ago) after trial and error I found my data in the USA. Im in Australia. They are really trying hard to make this overcomplicated.

**Comment by e-dupuis** [comment]
Still having the issue on the latest docker version

**Comment by virdb** [comment]
same here, still broken

**Comment by Tontze** [comment]
You need to have a device in IoT platform...
What helped for me was: 
Under "Devices" linked my Smartlife app to Tuya IoT platform, then under "all devces" added my Smartlife account and got my device listed.
After this, i got rid of the error and my devices popped up :)

**Comment by virdb** [comment]
I can confirm. My issue was the location of the devices of my linked Smart Life. 
I changed the datacenter (in the device section) and linked my account again.
Now the devices are presents and the integration works.

**Comment by nanoamp** [comment]
I didn't get any devices until I _first_ set the correct datacenter (which wasn't the obvious one from my geographical location - I needed to look-up the appropriate one [here](https://developer.tuya.com/en/docs/iot/oem-app-data-center-distributed?id=Kafi0ku9l07qb)) and _then_ linked my SmartLife account to the Tuya IoT platform via Devices > Link Tuya App Account. With the wrong datacenter selected, the SmartLife account would still link, but it wouldn't pick up any devices, and the HASS integration would throw this error.

**Comment by luixal** [comment]
So crazy! In case some gets here having the same problem I had:

- If you get the `skill id invalid` is totally related to geographical mismatch. I'm in Spain, so I picked Central Europe in the Tuya Iot Platform project. In HA, I had to pick my country, Spain. Once that done, the `skill id invalid` error went away, but the `permission deny` appeared.
- The `permission deny` problem was solved as soon as I used the username/email and password I used in my phone's Smart Life app. I was using the credentials created for the Tuya IoT Platform. Once I entered the credentials in the mobile app everything worked out!

Cheers!

**Comment by rileyz** [comment]
I had this error also. 

- For the second set of credentials, its the Tuya Life app username and password.
- Just to reiterate the geographical mismatch, check your location is correct with the URL below. I thought I was Western Europe, turns out I needed to set to Central Europe 🤦‍♂️

https://developer.tuya.com/en/docs/iot/oem-app-data-center-distributed?id=Kafi0ku9l07qb

**Comment by baylanger** [comment]
I might have found a "workaround" for this.

I am in Eastern Canada/US and ran today into this same exact "2406 skill id invalid" issue.

First, I followed the instructions [here](https://www.home-assistant.io/integrations/tuya/).  Installed Tuya Smart, created a tuya.com account created to add a Tuya Cloud Project (per instructions).  Since I'm on East coast, I picked Eastern America Data Center"

Then I tried to configure Tuya in hass and kept getting the "2406" error.  I scramble and went on a trial / error.  Then I stumble on [this](https://developer.tuya.com/en/docs/iot/oem-app-data-center-distributed?id=Kafi0ku9l07qb#title-7-Countries%20or%20regions%20supporting%20only%20certain%20data%20centers) Tuya document that shows some data centers are available only from specific regions... and I noticed "Eastern America DC" isn't even part of the list.

I went on tuya.com and added "Western America DC" to the Cloud Project.  BINGO, my devices finally showed up on tuya.com IoT portal!

Bottom line, if you're trying to use Eastern America Data Center, try to use the Western one.

**Comment by mschipperheyn** [comment]
From Brazil, Western American doesn't work. Even though it is listed in the docs: https://github.com/tuya/tuya-home-assistant/blob/main/docs/regions_dataCenters.md

**Comment by Quaziz-creator** [comment]
I give up! Nothing is working. I use other brands from now on

**Comment by piereligio** [comment]
GUYSGUYSGUYS solution found!
"If your cloud project was created before May 25, 2021, you need to perform project migration. ."
From here: https://developer.tuya.com/en/docs/iot/Home-assistant-tuya-intergration?id=Kb0eqjig0utdd
Basically you got to do a new project if your one was older than May 25. 
For doing the new project I followed:
https://developer.tuya.com/en/docs/iot/Platform_Configuration_smarthome?id=Kamcgamwoevrx
and this:
https://developer.tuya.com/en/docs/iot/Home-assistant-tuya-intergration?id=Kb0eqjig0utdd

**Comment by josepeinado** [comment]
@luixal I've tried to set it up for Spain. I selected Eastern Europe as the data center for my project and my smart life app is already linked but the same error appears: "Login failed: {'code': 2406, 'msg': 'skill id invalid'," when trying to install the Tuya V2 integration, any idea / suggestion? What's the literal code you entered: '34'? Thanks.

**Comment by luixal** [comment]
Hi @josepeinado , if you read my post, I selected "Central Europe", not Eastern. I don't know which literal I entered or if I entered any at all, I think it was just a dropdown. Anyhow, there's a table linked around for you to find you zone (Central Europe for Spain).

**Comment by teedohs** [comment]
Western and Eastern America worked for me and I'm in the Eastern US. Make sure that in the Tuya IoT project, you go to the Devices tab. After that, link you Tuya regular account so that all of the devices from your regular account show up in your IoT Tuya account. Finally, go back to Home Assistant and add the Tuya Integration. The Access ID and Token you'll add from the project in the Tuya IoT account. However, the username and password that you'll use for the integration will be the email address and password from the standard Tuya account and NOT the credentials from the Tuya IoT account.

**Comment by SleepWalker** [comment]
> GUYSGUYSGUYS solution found!
From documentation: "Your cloud project on the Tuya IoT Development Platform should be created after May 25, 2021. Otherwise, you need to create a new project."

this solved my issue. I have created my project a long time ago when I wasn't using HA and now tried to use it for HA. Recreating a new project have solved all the issues.

**Comment by dracomithril** [comment]
I had the same issue recently and to fix that I created new iot tuya project with `Device Status Notification` added just like in [documentation](https://www.home-assistant.io/integrations/tuya) and all started to work
